segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/hdfs/FileSystemOperation.java

```diff
@@ -193,7 +193,9 @@ FileDescriptor combine(FileDescriptor target, List<FileDescriptor> sources, bool
             }
 
             // Finally, concatenate the source files into the target.
-            this.context.fileSystem.concat(target.getPath(), toConcat);
+            if (toConcat.length > 0) {
```

Comment:
This may end up being wiped away completely with #2401 , but this became really bothersome during the testing of this PR.